XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -343,6 +354,7 @@ public int MinVCPUs()
                 return (T) value;
 
             var templateValues = GetRestrictionValueAcrossTemplates<T>(vm, field, "max");
+            templateValues.Add(defaultValue);
             return templateValues.Count == 0 ? defaultValue : templateValues.Max();

```

Comment:
surely templateValues will never be none if you just added to it. Same with line 377

Reply:
good point 😓